includes/blocks.inc

```diff
@@ -556,13 +555,20 @@ function islandora_solr_advanced_search_form($form, &$form_state) {
       'terms' => array(''),
     );
   }
+
   $terms = array(
     '#type' => 'markup',
     '#prefix' => '<div id="islandora-solr-advanced-terms">',
     '#suffix' => '</div>',
     '#tree' => TRUE,
   );
   foreach ($values['terms'] as $i => $value) {
+    if (variable_get('islandora_solr_advanced_search_block_lucene_syntax_escape', FALSE)) {
+      if (isset($value['search'])) {
+        $value['search'] = islandora_solr_restore_slashes($value['search']);
```

Comment:
I don't believe this should actually be necessary, assuming the query has actually been built... [One of the first things the query processor does is to run restore slashes](https://github.com/Islandora/islandora_solr_search/blob/b185566d4aa1d9cc28b68823988affea5c742f61/includes/query_processor.inc#L121).